### PR TITLE
Edit comments, fixes #653

### LINKED ISSUE #653: Edit Comments in Work Item Detail View
_From @michaelkleinhenz on October 12, 2016 8:39_

**Reference:** 1608E102/1608E063
# Description

As a user, I want to be able to edit my comments in the work item detail view.
# Functional Acceptance Criteria
1. The user can edit a comment by clicking on the edit button.
2. An edited comment is saved by clicking on the save button.
3. Comments can only be edited by the original author (user).
4. All updates are done immediately and in-place.
# Non-functional Acceptance Criteria

None.
# Dependencies
1. Userstory #298 
2. Userstory #348
3. Userstory #349
4. Userstory #350

_Copied from original issue: almighty/almighty-core#351_

### PR DESCRIPTION
**Pull request by debloper**
Simple implementation of comment editing for the original author of a comment.

- Comments added by the signed up user will be shown as editable text elements
- After editing, on `<Enter>` the comment is updated (`<Shift+Enter` to line-break)
- After updating takes focus off of comment. If focus wasn't reset, comment wasn't updated.

### CODE REVIEW

**Comment by michaelkleinhenz** [comment]
[test]

File: src/app/work-item/work-item-detail/work-item-comment/work-item-comment.component.html [hunk 1]
```diff
@@ -39,7 +39,19 @@ <h5 class="col-md-6 col-sm-6 col-xs-8 no-padding truncate">
                                       {{comment.relationalData?.creator?.attributes?.fullName}}
                                   </strong>
                               </h5>
-                               <p class="col-md-12 col-xs-12 col-sm-12 no-padding" id="{{'comment_body_' + counter}}">{{comment.attributes.body}}</p>
+                              <p *ngIf="!loggedIn || (currentUser.id !== comment.relationalData.creator?.id)"
+                                  id="{{'comment_body_' + counter}}"
+                                  class="col-md-12 col-xs-12 col-sm-12 no-padding">
+                                {{ comment.attributes.body }}
+                              </p>
+                              <p  *ngIf="!!loggedIn && (currentUser.id === comment.relationalData.creator?.id)"
```
Comment (nimishamukherjee) [inline src/app/work-item/work-item-detail/work-item-comment/work-item-comment.component.html]: This implementation varied from what UX has specified. 
Please create an issue which states that we need to match the UX (Kebab menu implementation).
Also, there is no indication that a comment is editable. Can we have a cursor icon for now?

**Comment by debloper** [comment]
# FUNCTIONAL TESTS ARE BROKEN
PR will be updated once it's fixed (ping me please)
Meanwhile, please review and approve if the code it alright for this PR

File: src/app/work-item/work-item-detail/work-item-comment/work-item-comment.component.html (hunk 1 above)
Comment (debloper) [inline src/app/work-item/work-item-detail/work-item-comment/work-item-comment.component.html]: UXD specified editable interface has many inconsistencies, so we made a functional implementation like this. For now, this should suffice for now.

@SMahil is working on some changes on comments UI and if we need more then we can have them implemented later when we have better clarity.

**Pull request merged**